pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go

```diff
@@ -34,7 +39,43 @@ import (
 
 func CreateHostDevices(vmi *v1.VirtualMachineInstance) ([]api.HostDevice, error) {
 	SRIOVInterfaces := vmispec.FilterSRIOVInterfaces(vmi.Spec.Domain.Devices.Interfaces)
-	return CreateHostDevicesFromIfacesAndPool(SRIOVInterfaces, NewPCIAddressPool(SRIOVInterfaces))
+
+	pciAddressPoolWithNetworkStatus, err := newPCIAddressPoolWithNetworkStatusFromFile(sriov.MountPath + "/" + sriov.VolumePath)
+	if err != nil {
+		return nil, err
+	}
+	if pciAddressPoolWithNetworkStatus.Len() == 0 {
+		log.Log.Object(vmi).Warningf("found no SR-IOV networks to PCI-Address mapping. fall back to resource address pool")
+		return CreateHostDevicesFromIfacesAndPool(SRIOVInterfaces, NewPCIAddressPool(SRIOVInterfaces))
+	}
+
+	return CreateHostDevicesFromIfacesAndPool(SRIOVInterfaces, pciAddressPoolWithNetworkStatus)
+}
+
+func newPCIAddressPoolWithNetworkStatusFromFile(path string) (*PCIAddressWithNetworkStatusPool, error) {
```

Comment:
I think this logic here is not trivial and should have been explained in the function description.
Explicitly, when an error is returned, when no error is returned but also no pool.

A common reader who is not in deep details will have hard time understanding the intention here and the logic of when to fail and when not.

Reply:
Added function documentation, I agree it's not trivial.

Also temporarily changed it to public to make sure the Len returns 0 in case we want to fallback
```
	It("CHECK CHECK", func() {
		pool, err := sriovhostdev.RAMnewPCIAddressPoolWithNetworkStatusFromFile("/sdf/sdf")

		Expect(err).ToNot(HaveOccurred())
		Expect(pool.Len()).To(BeZero())
	})
```

passed